server/api/collections.js

```diff
@@ -571,6 +574,27 @@ router.post("collections.list", auth(), pagination(), async (ctx) => {
     limit: ctx.state.pagination.limit,
   });
 
+  const nullIndexCollection = collections.findIndex(
+    (collection) => collection.index === null
+  );
+
+  const indexSet = new Set();
+
+  const repeatedIndex = collections.some((collection) => {
+    if (indexSet.has(collection.index)) {
+      return true;
+    }
+    indexSet.add(collection.index);
+    return false;
+  });
+
```

Comment:
I'm not a huge fan of this much logic running in the controller method, is the repeated index check necessary?

I was previously thinking the "migration" would be a one-time thing, perhaps the repeat check should happen as part of `collections.move` before writing the new index?

Reply:
I agree, let's move this logic elsewhere. 

I think the scenario which will lead to the repeated index is when a private collection is made public. 

How about we run this logic to specifically when the visibility of collections is changed? 

I think It will be unpredictable for the user when they move a collection `X` and they see other collections move as well because of index collision.